lib/crypto/sr25519/sr25519.go

```diff
@@ -43,6 +44,20 @@ type PrivateKey struct {
 	key *sr25519.SecretKey
 }
 
+// SignVerify verify signature given pubkey and msg
+func SignVerify(pubkey, sig, msg []byte) (bool, error) {
+	pubKey, err := NewPublicKey(pubkey)
+	if err != nil {
+		return false, fmt.Errorf("failed to fetch sr25519 public key: %s", err)
+	}
+	ok, err := pubKey.Verify(msg, sig)
+	if err != nil || !ok {
+		return false, fmt.Errorf("failed to verify sr25519 signature: %s", err)
```

Comment:
Please split this if condition; we want to have one checking the error and returning `return false, fmt.Errorf("failed to verify signature: %w", err)` and another condition `if !ok` returning `return false, nil`

Otherwise there is no point returning two values `ok` and `err` from this function